rollup/sync_service.go

```diff
@@ -258,384 +142,94 @@ func (s *SyncService) Start() error {
-	if err != nil {
-		return fmt.Errorf("Error binding to contracts: %w", err)
-	}
-	// Check the sync status of the eth1 node
-	err = s.checkSyncStatus()
-	if err != nil {
-		return fmt.Errorf("Bad sync status: %w", err)
-	}
-
-	// Set the initial values of the `LatestL1BlockNumber`
-	// and `LatestL1Timestamp`
-	err = s.initializeLatestL1()
-	if err != nil {
-		return fmt.Errorf("Cannot set latest L1: %w", err)
-	}
-
-	go s.LogDoneProcessing()
-	// Catch up to the tip of the eth1 chain
-	err = s.processHistoricalLogs()
-	if err != nil {
-		return fmt.Errorf("Cannot process historical logs: %w", err)
+		s.setSyncStatus(false)
 	}
 
-	gasLimit, err := s.mgrCaller.GetMaxTransactionGasLimit(&bind.CallOpts{
-		Context: s.ctx,
-	})
-	if err != nil {
-		return fmt.Errorf("Cannot fetch gas limit: %w", err)
-	}
-	s.gasLimit = gasLimit.Uint64()
-	log.Info("Setting max transaction gas limit", "gas limit", s.gasLimit)
-
 	go s.Loop()
-	go s.pollHead()
-	go s.ClearTransactionLoop()
-
-	if !s.verifier {
-		go s.sequencerIngestQueue()
-	}
-
 	return nil
 }
 
 // initializeLatestL1 sets the initial values of the `L1BlockNumber`
 // and `L1Timestamp` to the deploy height of the Canonical Transaction
 // chain if the chain is empty, otherwise set it from the last
-// transaction processed.
-func (s *SyncService) initializeLatestL1() error {
+// transaction processed. This must complete before transactions
+// are accepted via RPC when running as a sequencer.
+func (s *SyncService) initializeLatestL1(ctcDeployHeight *big.Int) error {
 	block := s.bc.CurrentBlock()
 	if block == nil {
 		return errors.New("Current block is nil")
 	}
 	if block == s.bc.Genesis() {
-		if s.ctcDeployHeight == nil {
+		if ctcDeployHeight == nil {
 			return errors.New("Must configure with canonical transaction chain deploy height")
 		}
-		var err error
-		block, err = s.ethclient.BlockByNumber(s.ctx, s.ctcDeployHeight)
+		context, err := s.client.GetEthContext(ctcDeployHeight.Uint64())
 		if err != nil {
-			return fmt.Errorf("Cannot fetch ctc deploy block at height %d", s.ctcDeployHeight)
+			return fmt.Errorf("Cannot fetch ctc deploy block at height %d: %w", ctcDeployHeight.Uint64(), err)
 		}
-		s.SetLatestL1Timestamp(block.Time())
-		s.SetLatestL1BlockNumber(block.Number().Uint64())
+		s.SetLatestL1Timestamp(context.Timestamp)
+		s.SetLatestL1BlockNumber(context.BlockNumber)
+		// There has yet to be any enqueued transactions
+		// TODO: note- when querying enqueue txs, make sure to not double play
+		s.SetLatestEnqueueIndex(0)
 	} else {
 		txs := block.Transactions()
 		if len(txs) != 1 {
 			log.Error("Unexpected number of transactions in block: %d", len(txs))
 		}
-		if len(txs) > 0 {
-			tx := txs[0]
-			s.SetLatestL1Timestamp(tx.L1Timestamp())
-			s.SetLatestL1BlockNumber(tx.L1BlockNumber().Uint64())
+		tx := txs[0]
+		s.SetLatestL1Timestamp(tx.L1Timestamp())
+		s.SetLatestL1BlockNumber(tx.L1BlockNumber().Uint64())
+		// Set the last enqueue index for the sequencer
```

Comment:
@kfichter This is the logic for walking backwards from the latest enqueue'd transaction to the latest enqueue'd transaction that has been included in the ctc. This logic works but is not ideal, it would be better to have some sort of passthrough to the ctc calls `getPendingQueueElements()`, `getNumQueueElements()`, `getNextQueueElement()`. Not sure if indexing this information is easier than allowing a passthrough because the passthrough would require an archive node. If it is simple to index then that could be a solution to simplify this logic that involves a linear search